### PR TITLE
Add sample metadata categories to clustergrammer visualization and normalize NPX data

### PR DESCRIPTION
**Pull request by jacoblurye**

### CODE REVIEW

File: functions/visualizations.py [hunk 1]
```diff
@@ -35,15 +36,42 @@ def vis_preprocessing(event: dict, context: BackgroundContext):
...
+def _get_metadata_df(trial_id: str) -> pd.DataFrame:
+    """
+    Build a dataframe containing the participant/sample metadata for this trial,
+    joined on CIMAC ID and indexed on CIMAC ID.
+    """
+    participants_blob = _get_data_file(f"{trial_id}/participants.csv", as_string=True)
+    samples_blob = _get_data_file(f"{trial_id}/samples.csv", as_string=True)
+
+    participants_df = pd.read_csv(participants_blob)
+    samples_df = pd.read_csv(samples_blob)
```
Comment (curlup) [inline functions/visualizations.py]: just a thought - we also have this data in the json format in the DB, so maybe it's faster to get it from there than from GCS?

File: functions/visualizations.py [hunk 2]
```diff
@@ -35,15 +36,42 @@ def vis_preprocessing(event: dict, context: BackgroundContext):
...
+
+    metadata_df = pd.merge(
```
Comment (curlup) [inline functions/visualizations.py]: I'm not sure, but my understanding was that samples file includes everything from the participants csv. It isn't?

File: functions/visualizations.py (hunk 1 above)
Comment (jacoblurye) [reply functions/visualizations.py]: Reviewing the `patient_sample_update.generate_csvs` cloud function where these files are created, we don't appear to be storing the contents of these files as JSON in the database - only metadata about them: https://github.com/CIMAC-CIDC/cidc-cloud-functions/blob/d72a54989fe5939d191b8f2da69441fce13b90f6/functions/patient_sample_update.py#L31

File: functions/visualizations.py (hunk 2 above)
Comment (jacoblurye) [reply functions/visualizations.py]: The participants CSV contains a 'cohort_name' column, which the samples CSV does not have in it. Maybe we should just be creating a single flattened samples CSV for each trial though.

File: functions/visualizations.py (hunk 1 above)
Comment (curlup) [reply functions/visualizations.py]: I meant that we have the JSON that we generated this csv from, so wouldn't it be faster to call `generate` on the fly rather then reading csv from gcs

Comment (jacoblurye) [reply functions/visualizations.py]: Ah, I see - almost definitely. We can put a ticket in the backlog about this.

**Pull request merged**